NARFO_BE/NARFO_BE/Controllers/PendingRenewalsController.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using NARFO_BE.Models;
+
+namespace NARFO_BE.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class PendingRenewalsController : ControllerBase
+    {
+        private readonly narfoContext _context;
+
+        public PendingRenewalsController(narfoContext context)
+        {
+            _context = context;
+        }
+
+        // GET: api/PendingRenewals
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<PendingRenewal>>> GetPendingRenewal()
+        {
+            return await _context.PendingRenewal.ToListAsync();
+        }
+
+        // GET: api/PendingRenewals/5
+        [HttpGet("{id}")]
+        public async Task<ActionResult<PendingRenewal>> GetPendingRenewal(int id)
+        {
+            var pendingRenewal = await _context.PendingRenewal.FindAsync(id);
```

Comment:
is there a reason you have so many asynchronous calls. what benefit does this give you? you have no code running bellow this that has to asynchronously wait for this calls response. this would be more relevant when you execute stored procedures or time consuming database calls.

Reply:
you are also not running any parallel processing on these api's so this is a waste of resources